src/formatDate.js

```diff
@@ -50,7 +53,66 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const YEAR_FORMAT_YY = 'YY';
+  const YEAR_FORMAT_YYYY = 'YYYY';
+  const DAY_FORMAT = 'DD';
+  const MONTH_FORMAT = 'MM';
+  const getPrefixYear = (yearBefore, newFormat) => {
+    const yearUpdateFormat
+      = newFormat.find(partDate => partDate.includes(YEAR_FORMAT_YY));
+
+    if (yearBefore.length === yearUpdateFormat.length) {
+      return yearBefore;
+    }
+
+    const PREVIOUS_CENTURY = '19';
+    const CURRENT_CENTURY = '20';
+    const TO_DETERMINE_CENTURY = 30;
+
+    if (yearBefore.toString().length > yearUpdateFormat.length) {
+      return yearBefore.slice(-2);
+    }
+
+    if (yearUpdateFormat.length === 4) {
+      const century = yearBefore < TO_DETERMINE_CENTURY
+        ? CURRENT_CENTURY
+        : PREVIOUS_CENTURY;
+
+      return century + yearBefore;
+    }
+  };
+
+  const newFormatDate = [];
+  const SPLITTED_OLD_DATE = date.split(fromFormat[fromFormat.length - 1]);
+  const indexYY = fromFormat
+    .findIndex((partDate) =>
+      partDate === YEAR_FORMAT_YY || partDate === YEAR_FORMAT_YYYY);
+
+  const indexForMounth = fromFormat.indexOf(MONTH_FORMAT);
+  const indexForDay = fromFormat.indexOf(DAY_FORMAT);
+
+  const day = SPLITTED_OLD_DATE[indexForDay];
+  const month = SPLITTED_OLD_DATE[indexForMounth];
+  const year = SPLITTED_OLD_DATE[indexYY];
+  const SEPARATOR_INDEX = toFormat[toFormat.length - 1];
```

Comment:
SEPARATOR_INDEX and SPLITTED_OLD_DATE are not general constants as they depend on what you pass to function. So use camelCase for them instead.